validation/jinja2/validation/segment_details.html

```diff
@@ -1,8 +1,53 @@
+<script>
+    function showDiv(name, suggestions=null) {
+        document.getElementById('edit').style.display = "block";
+        document.getElementById('id_cree').focus()
+        analysis = '';
+
+        if (suggestions == null) {
+            translation = document.getElementById(name + '-translation').innerHTML;
+        }
+
+        else {
+            translation = suggestions[name]['translation'];
+            analysis = suggestions[name]['analysis'];
+        }
+
+        document.getElementById('id_cree').value = name;
+        document.getElementById('id_transl').value = translation;
+        document.getElementById('id_analysis').value = analysis;
+    }
+
+    function hideDiv() {
+        document.getElementById('edit').style.display = "none";
+    }
+
+
+    function revert(transcription, translation, analysis) {
+        document.getElementById('edit').style.display = "block";
+        document.getElementById('id_cree').focus()
+
+        document.getElementById('id_cree').value = transcription;
+        document.getElementById('id_transl').value = translation;
+        document.getElementById('id_analysis').value = analysis;
+    }
+
+    
+</script>
+
 {% extends 'validation/_base.html' %}
 {% import 'validation/_macros.html' as macros %}
 
 {% block content %}
 <div class="table-responsive">
+  {% if auth %}
+    <button data-cy="edit-button" class="favorite styled"
+        onclick="showDiv('{{ segment_name }}')"
+        style="float: right">
```

Comment:
Extract to its own style?

Also, I'm starting to get in the habbit of using `display: flex; flex-direction: row` in the parent and either also put `align-items: end` in the flex container or add `align-self: end` to the current element. Because floats have historically had MANY issues and hacks associated in their usage for layouts like this.

Reply:
I can for sure do this. I'll be going through and looking for all inline styles in my final PR for the properly implemented style sheet.

Reply:
Good point. I learned on `float` and `clearfix` because it was all we had back then, but whenever I find myself using them now, after somewhere between 1 and 100 minutes I realize that if I’d used flex from the beginning I would have been done, with better spacing, in about a minute.